src/contracts/core/DelegationManager.sol

```diff
@@ -289,27 +289,21 @@ contract DelegationManager is
     function decreaseDelegatedShares(
         address staker,
         uint256 curDepositShares,
-        uint64 prevBeaconChainSlashingFactor,
-        uint256 wadSlashed
+        uint64 beaconChainSlashingFactorDecrease
     ) external onlyEigenPodManager {
         if (!isDelegated(staker)) {
             return;
         }
-
         address operator = delegatedTo[staker];
 
-        // Calculate the previous slashing factor. This mirrors the calculation done
-        // in _getSlashingFactor
+        // Calculate the shares to remove from the operator by calculating difference in shares
+        // from the newly updated beaconChainSlashingFactor
         uint64 maxMagnitude = allocationManager.getMaxMagnitude(operator, beaconChainETHStrategy);
-        uint256 prevSlashingFactor = maxMagnitude.mulWad(prevBeaconChainSlashingFactor);
-
-        // Calculate the shares to remove from the operator by calculating the previously-withdrawable
-        // shares and applying slashing
         DepositScalingFactor memory dsf = _depositScalingFactor[staker][beaconChainETHStrategy];
         uint256 sharesToRemove = dsf.calcWithdrawable({
             depositShares: curDepositShares,
-            slashingFactor: prevSlashingFactor
-        }).mulWad(wadSlashed);
+            slashingFactor: maxMagnitude.mulWad(beaconChainSlashingFactorDecrease)
+        });
```

Comment:
So IIUC, this is equivalent to the old calculation.

Old calculation is `depositShares*maxMag*prevFactor*wadSlashed`
New Calculation is `depositShares*maxMag*bsfDecrease`

Must be the case that `prevFactor*wadSlashed = bsfDecrease`

Splitting these up,

`pF*WS = pF - nF`, where pF is prevBeaconChainScalingFactor, WS is wadSlashed, and nf Is newBeaconChainScalingFactor.

`WS = -nF/pF`

Reply:
Old calculation is actually `depositShares*dsf*maxMag*wadSlashed`
because its using depositScalingFactor as well and prevSlashingFactor in the old code was just the maxMagnitude itself. 
New calculation is now `depositShares*dsf*maxMag*bsfDecrease`

Implying wadSlashed and bsfDecrease are equivalent which intuitively makes sense.

Reply:
Update: @ypatil12 is correct here actually, the prevBeaconChainSlashingFactor is used in the old calculation and
`prevFactor*wadSlashed = bsfDecrease` is true